test/rulesets/Base/fastmath_able.jl

```diff
@@ -60,7 +60,7 @@ const FASTABLE_AST = quote
         @testset "Multivariate" begin
             @testset "sincos(x::$T)" for T in (Float64, ComplexF64)
                 x, Δx, x̄ = randn(T, 3)
-                Δz = (randn(T), randn(T))
+                Δz = Composite{Tuple{T,T}}(randn(T), randn(T))
```

Comment:
This worked without this, because we ducktype on any iterable.
But `Composite{Tuple{T, T}}` is the right type

Reply:
Just making sure I understand, [this is the line](https://github.com/JuliaDiff/ChainRulesCore.jl/blob/master/src/rule_definition_tools.jl#L180) that ducktypes right?

Reply:
How did you find this actually?

Reply:
This is the line that ducktypes
https://github.com/JuliaDiff/ChainRulesCore.jl/blob/95b9fd72824945374bd1767a0fcc380265a7a038/src/rule_definition_tools.jl#L185
Because it basically generates `pullback((arg1, arg2))` rather than `pullback(x::Composite)` or `pullback((arg1, arg2)::Composite)`
It is just relying on destructuing, which just requires iteration.

> How did you find this actually?

When I was debugging the issue in https://github.com/JuliaDiff/ChainRulesCore.jl/pull/276
I just noticed it in passing